src/main/java/chalmers/pimp/model/canvas/layer/Doodle.java

```diff
@@ -0,0 +1,157 @@
+package chalmers.pimp.model.canvas.layer;
+
+import chalmers.pimp.model.IRenderer;
+import chalmers.pimp.model.color.IReadOnlyColor;
+import chalmers.pimp.model.pixeldata.IPixel;
+import chalmers.pimp.model.pixeldata.IReadOnlyPixelData;
+import chalmers.pimp.model.pixeldata.PixelData;
+import chalmers.pimp.model.pixeldata.PixelFactory;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A layer which has a list of points and draws straight lines between the points.
+ */
+final class Doodle implements ILayer {
+
+  private final List<IPixel> pixels;
+  private final LayerDelegate layerDelegate;
+  private final IReadOnlyColor color;
+  private final int lineWidth;
+  private int width;
+  private int height;
+
+  Doodle(int lineWidth, IReadOnlyColor color) {
+    pixels = new ArrayList<>();
+    layerDelegate = new LayerDelegate(LayerType.DOODLE);
+    layerDelegate.setName("Doodle");
+    this.lineWidth = lineWidth;
+    this.color = color;
+    width = 0;
+    height = 0;
+  }
+
+  @Override
+  public void setPixel(IPixel pixel) {
+    pixels.add(pixel);
+    if (pixel.getX() > width - 1) {
+      width = pixel.getX() + 1;
+    }
+    if (pixel.getY() > height - 1) {
+      height = pixel.getY() + 1;
+    }
```

Comment:
Is it correct to set the width and height the pixel coordinate and add 1?

Reply:
I added 1 because the coordinates are 0-indexed but the width is not.

I just recognized that this is incorrect because the layer width should be "largest x" + "line diameter"/2. Say the user draws a line from (0,0) to (0,100) then the layer is "line diameter" wide.